mass-deployer/internal/parser/parser.go

```diff
@@ -0,0 +1,89 @@
+package parser
+
+import (
+	"fmt"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+)
+
+func ParseConfig(configFile []byte) (Config, error) {
+	conf := Config{}
+
+	err := yaml.Unmarshal(configFile, &conf)
+	if err != nil {
+		return Config{}, err
+	}
+
+	if len(conf.NodeGroups) == 0 {
+		return Config{}, fmt.Errorf("couldn't find any nodes groups to use in deployment")
+	}
+	if len(conf.Vms) == 0 {
+		return Config{}, fmt.Errorf("couldn't find any vms that need to be deployed")
+	}
+	if len(conf.SSHKeys) == 0 {
+		return Config{}, fmt.Errorf("user ssh_keys is invalid, ssh_keys shouldn't be empty")
+	}
+	if len(conf.Mnemonic) == 0 {
+		return Config{}, fmt.Errorf("user mnemonics is invalid, mnemonics shouldn't be empty")
+	}
+	networks := []string{"dev", "test", "qa", "main"}
+	if !slices.Contains(networks, conf.Network) {
+		return Config{}, fmt.Errorf("netwok %s is invalid", conf.Network)
+	}
+
+	for _, nodeGroup := range conf.NodeGroups {
+		if nodeGroup.Name == "" {
+			return Config{}, fmt.Errorf("node groups name is invalid, shouldn't be empty")
+		}
+		if nodeGroup.NodesCount <= 0 {
+			return Config{}, fmt.Errorf("nodes_count in node_group: %s is invalid, should be a positive number", nodeGroup.Name)
+		}
+		if nodeGroup.FreeCPU <= 0 {
+			return Config{}, fmt.Errorf("free_cpu in node_group: %s is invalid, should be a positive number", nodeGroup.Name)
+		}
+		if nodeGroup.FreeMRU <= 0 {
+			return Config{}, fmt.Errorf("free_mru in node_group: %s is invalid, should be a positive number", nodeGroup.Name)
+		}
+	}
+
+	for _, vm := range conf.Vms {
+		if vm.Name == "" {
+			return Config{}, fmt.Errorf("vms name is invalid, shouldn't be empty")
+		}
+		if vm.Count <= 0 {
```

Comment:
check if the count doesn't match the count of nodes group

Reply:
The vms count can be not the same as nodes count I guess, I mean more than one vm can be deployed on the same node, right?